Add compatibility for other plugins parsing top-level returns in Astro frontmatter

FugiTech opened the pull request:
## Changes

Fixes #326 

Before, prettier-plugin-astro would pass frontmatter as-is to prettier's typescript formatter. This would break plugins that parse the typescript if there are any top-level returns in the frontmatter, as top-level returns are not valid javascript.

This change 
- adds a marker before all `return` statements in Astro frontmatter
- changes `return` to `throw`
- runs the typescript prettier on the modified frontmatter
- finds the marker to turn the `throw`s back to `return`s
- removes the marker
- cleans up any comments that were modified
 
It's a pretty ugly hack and I'm not proud of it, but it seemingly gets the job done. I expect there is some input on which it will break horribly.

As a note, I did try using a comment as a marker but that breaks if a user uses "return" as a word in comments, which is very likely. I also tried putting the marker after the `return` instead of before, but found that it changed the formatted output in a way that is hard to safely reverse.

## Testing

I ran the existing tests. I also tested it on a variety of input and options manually with the `@trivago/prettier-plugin-sort-imports` plugin enabled to ensure the transforms worked as expected.

I did not add any tests for this feature as I wanted to first verify this feature is desired. If you like the idea and want me to add tests I would be happy to do so.

## Docs

No docs were updated as this is a compatibility fix and should not affect how end-users use the plugin.

Princesseuh commented:
I'm fine with this overall! (Prettier plugins are very often hack-y mess). Would love to see tests testing a few inputs to make sure it works in all cases. 

Producing broken output is a very big no-no in formatting, so it's important that it works perfectly

FugiTech commented:
Sorry for the delay, I've added a test with a variety of conditions in which the `return` keyword would be used at the top-level, not at the top-level, and in non-javascript contexts (comments, strings, html). In doing so I found some edge-cases and had to slightly change the code to use `___astro_return` (as a variable) rather than `"___astro_return"` (as a string). I believe this should not cause any issues.

I also changed the logic for how `___astro_return` markers are found to be more reliable, as the old way was accidentally removing comments above the return.

Princesseuh reviewed (approved):
Looks great, thank you for the tests!
